diff --git a/mkl/__init__.py b/mkl/__init__.py
index d2fefc7..47f35b9 100644
--- a/mkl/__init__.py
+++ b/mkl/__init__.py
@@ -23,7 +23,11 @@
 # OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 # OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
+import contextlib
+import os
 import sys
+from importlib.metadata import PackageNotFoundError, files
+
 
 class RTLD_for_MKL():
     def __init__(self):
@@ -38,6 +42,20 @@ class RTLD_for_MKL():
             sys.setdlopenflags(self.saved_rtld | ctypes.RTLD_GLOBAL)
         except AttributeError:
             pass
+
+        with contextlib.suppress(
+            AttributeError, FileNotFoundError, StopIteration, PackageNotFoundError
+        ):
+            if os.name == "nt":
+                # Add MKL library to DLL search path on Windows.
+                dll = next(p for p in files("mkl") if p.match("*mkl_rt*.dll"))
+                dll_dir = dll.locate().resolve().parent
+                os.add_dll_directory(dll_dir)
+            else:
+                # Preload MKL library on Linux.
+                dll = next(p for p in files("mkl") if p.match("*libmkl_rt.so*"))
+                ctypes.CDLL(dll.locate().resolve(), mode=ctypes.RTLD_GLOBAL)
+
         del ctypes
 
     def __exit__(self, *args):
