diff --git a/mkl/__init__.py b/mkl/__init__.py
index 4a86a64..1ebe93a 100644
--- a/mkl/__init__.py
+++ b/mkl/__init__.py
@@ -23,7 +23,10 @@
 # OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 # OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
+import contextlib
+import os
 import sys
+from importlib.metadata import PackageNotFoundError, files
 
 
 class RTLD_for_MKL:
@@ -41,6 +44,21 @@ class RTLD_for_MKL:
             sys.setdlopenflags(self.saved_rtld | ctypes.RTLD_GLOBAL)
         except AttributeError:
             pass
+
+        with contextlib.suppress(
+            AttributeError, FileNotFoundError, StopIteration, PackageNotFoundError
+        ):
+            if os.name == "nt":
+                # Add MKL library path to DLL search path on Windows.
+                dll = next(p for p in files("mkl") if p.match("*mkl_rt*.dll"))
+                dll_dir = dll.locate().resolve().parent
+                os.add_dll_directory(dll_dir)
+            else:
+                # There's no easy way to expand LD_LIBRARY_PATH at runtime in
+                # Linux. Instead preload MKL library directly.
+                dll = next(p for p in files("mkl") if p.match("*libmkl_rt.so*"))
+                ctypes.CDLL(dll.locate().resolve(), mode=ctypes.RTLD_GLOBAL)
+
         del ctypes
 
     def __exit__(self, *args):
