diff --git a/mkl/__init__.py b/mkl/__init__.py
index c0eb2ae..314cb21 100644
--- a/mkl/__init__.py
+++ b/mkl/__init__.py
@@ -31,8 +31,11 @@ class RTLD_for_MKL:
         self.saved_rtld = None
 
     def __enter__(self):
+        import contextlib
         import ctypes
+        import os
         import sys
+        from importlib.metadata import PackageNotFoundError, files
 
         try:
             self.saved_rtld = sys.getdlopenflags()
@@ -42,6 +45,21 @@ class RTLD_for_MKL:
             sys.setdlopenflags(self.saved_rtld | ctypes.RTLD_GLOBAL)
         except AttributeError:
             pass
+
+        with contextlib.suppress(
+            AttributeError, FileNotFoundError, StopIteration, PackageNotFoundError
+        ):
+            if os.name == "nt":
+                # Add MKL library path to DLL search path on Windows.
+                dll = next(p for p in files("mkl") if p.match("*mkl_rt*.dll"))
+                dll_dir = dll.locate().resolve().parent
+                os.add_dll_directory(dll_dir)
+            else:
+                # There's no easy way to expand LD_LIBRARY_PATH at runtime in
+                # Linux. Instead preload MKL library directly.
+                dll = next(p for p in files("mkl") if p.match("*libmkl_rt.so*"))
+                ctypes.CDLL(dll.locate().resolve(), mode=ctypes.RTLD_GLOBAL)
+
         del ctypes
 
     def __exit__(self, *args):
